solr/solr_setup.sh

```diff
@@ -17,34 +17,33 @@ ls -lart /var/solr/data/ckan/data;
 find $lockpath -name write.lock -delete;
 rm -rf $flagfile;
 
-# Run a check in the background for 10 mins.
-check_folder() {
-  # check the index folders and lock status
-  find ${lockpath} -type d -name index* | xargs du -ch
-  find $lockpath -name write.lock | xargs ls -l
-  echo "#### start index.properties"
-  [[ -f /var/solr/data/ckan/data/index.properties ]] && cat /var/solr/data/ckan/data/index.properties
-  echo "#### end index.properties"
+logfile=/var/solr/logs/solr.log
+# add a marker to the end of the log file
+if [ -f "$logfile" ]; then
+  echo "#### CORECHECK_MARKER" >> $logfile
+fi
+
+check_corelock() {
+  # check LockObtainFailedException since last check marker
+  awk -v RS='(^|\n)CORECHECK_MARKER"\n' 'END{printf "%s", $0}' $logfile | grep -q "LockObtainFailedException"
+  if [ $? -eq 0 ]; then
+    echo "Found LockObtainFailedException in log file. Quit solr."
+    solr stop -p 8983
```

Comment:
Will `root` be able to stop solr if it's being run from the `solr` user?

Reply:
Also, it seems like we need to add another marker if we find the exception too?  If we don't, will the process keep exiting?

![image](https://github.com/GSA/catalog.data.gov/assets/85196563/21bf3ddc-cbaa-4e80-bccd-16a3771fcaa1)

Reply:
I think we need something like... I don't think root has the `solr` command registered...
```suggestion
    su -c "solr stop -p 8983" -m solr
```

Reference: https://github.com/GSA-TTS/datagov-brokerpak-solr/blob/main/terraform/ecs/provision/leader.tf#L62-L63

Reply:
@FuhuXia verified `root` has the privilege.